packages/react/src/index.js

```diff
@@ -10,49 +10,49 @@ const createCss = (init) => {
 	const sheet = createCoreCss(init)
 
 	return assign(sheet, {
-		/** Returns a React component. */
+		/** Returns a React component in the form of a forwarded ref object. */
 		styled: (
 			/** Type of component. */
 			...inits
 		) => {
-			const defaultType = inits.map((init) => (Object(init).type ? init.type : init)).find((init) => init) || 'span'
+			const defaultType = inits.map((init) => (Object(init).stitchesType ? init.stitchesType : init)).find((init) => init) || 'span'
 			const composition = sheet.css(...inits.filter((init) => $$composers in Object(init) || (init && typeof init === 'object' && !init.$$typeof)))
 
-			/** Returns a React element. */
-			return Object.setPrototypeOf(
-				{
-					render(
-						/** Props used to determine the expression of the current component. */
-						initProps,
-						ref,
-					) {
-						// express the component, extracting `props`, `as` & `ref`
-						const {
-							props: { as: type = defaultType, ...props },
-							...expressedProps // eslint-disable-line no-unused-vars
-						} = composition(initProps)
+			/** This is a React component in the form of a forwarded ref object, with a few extra Stitches properties. */
+			return {
+				/** The render function on the forwarded ref object returns a React element. */
+				render(
+					/** Props used to determine the expression of the current component. */
+					initProps,
+					ref,
+				) {
```

Comment:
It might be a good idea to replace the manually created forwarded ref object with a call to `React.forwardRef((initProps, ref) => {...}` and then add the Stitches properties to the returned object. This would probably be less brittle since manually creating the forwarded ref object is not part of the React api.

Reply:
This would probably also help with supporting Preact #454